shared/test/dataDriver_test.js

```diff
@@ -151,6 +151,19 @@ describe("loop.DataDriver", () => {
     });
   });
 
+  describe("#handleEvent.error", () => {
+    it("should retry the connection on error", () => {
+        // This calls _connectToRoom, so don't stub it out until after.
+        driver.setupWindowData(new actions.SetupWindowData({
+          roomToken: "fakeSetup"
+        }));
+        sandbox.stub(driver, "_connectToRoom");
+        driver.handleEvent({ type: "error" });
+        clock.tick(driver.INITIAL_RETRY_TIMEOUT + 1);
+        sinon.assert.called(driver._connectToRoom);
+    });
```

Comment:
Given that we've implemented backoff, we should add a second test that makes sure the backoff timer is doubled. This first test should probably check that `_retryTimer` is doubled. The second test should check that it's quadrupled after a double error.